src/controller/python/chip/clusters/ClusterObjects.py

```diff
@@ -275,3 +275,59 @@ def _cluster_object(cls) -> ClusterObject:
                                   )
                               ],
                               bases=(ClusterObject,))
+
+
+class ClusterEventDescriptor:
+    '''
+    The ClusterEventDescriptor is used for holding an event's metadata like its cluster id, event id and its type.
+
+    Users should not initialize an object based on this class. Instead, users should pass the subclass objects to tell some methods what they want.
+
+    The implementation of this functions is quite tricky, it will create a cluster object on-the-fly, and use it for actual encode / decode routine to save lines of code.
+    '''
+    @classmethod
+    def ToTLV(cls, tag: Union[int, None], value):
+        writer = tlv.TLVWriter()
+        wrapped_value = cls._cluster_object(Value=value)
+        cls.event_type.PutFieldToTLV(tag,
+                                     asdict(wrapped_value)['Value'], writer, '')
+        return writer.encoding
+
+    @classmethod
+    def FromTLV(cls, tlvBuffer: bytes):
```

Comment:
Why does this need it's own `FromTLV` methods? Why cant it just the versions int he base class?